Classes/ExpressionLanguage/DocumentTypeFunctionProvider.php

```diff
@@ -162,6 +161,7 @@ protected function loadDocument(array $requestData, int $pid): void
         if (!empty($requestData['id'])) {
             $this->initializeRepositories($pid);
             $doc = null;
+            $this->document = null;
```

Comment:
Is this really necessary? `$this->document` is declared as a class property and should be `null` anyways as long as it is not explicitly set. Could you explain your reasoning behind this change?

Reply:
Reason was a NPE or not existing/declarated `$this->document` at following point https://github.com/kitodo/kitodo-presentation/blob/e0b76f6ecd88771cad79a0d95b19dc565a787e02/Classes/ExpressionLanguage/DocumentTypeFunctionProvider.php#L189